File: 2025/t1/t1.py
with open('inp_full.txt', 'r') as f:
# with open('inp_base.txt', 'r') as f:
    lines = f.readlines()

def part1(lines):
    start = 50
    zeros = 0
    for l in lines:
        mult = 1
        if l[0] == 'L':
            mult = -1
        smth = int(l[1:]) % 100
        start += mult * smth
        # print(f'{prev} + {l.strip()} = {start}', end='')
        if start < 0:
            start = 100 + start
        elif start > 99:
            start = start - 100

        if start == 0:
            zeros += 1
        # print(f'({start}); zeros: {zeros}')

    print(zeros)

def part2(lines):
    start = 50
    zeros = 0
    for l in lines:
        mult = 1
        if l[0] == 'L':
            mult = -1
        prev = start
        clicks = int(l[1:])
        zeros += clicks // 100
        smth =  clicks % 100

        start += mult * smth
        # print(f'{prev} + {l.strip()} = {start}', end='')

        if prev != 0 and (start <= 0 or start > 99):
            zeros += 1
        if start < 0:
            start = 100 + start
        elif start > 99:
            start = start - 100
        # print(f'({start}); zeros: {zeros}')
        # if clicks > 100:
            # print(f"+ {clicks // 100} rotations")

    print(zeros)

part1(lines)
part2(lines)

File: 2025/t1/t1.ts
import * as fs from 'fs';

const lines = fs.readFileSync('inp_base.txt', 'utf-8').trim().split('\n');
const lines2 = fs.readFileSync('inp_full.txt', 'utf-8').trim().split('\n');

function part1(lines: string[]): void {
    let position = 50;
    let zeros = 0;

    for (const l of lines) {
        let mult = (l[0] === 'L') ? -1 : 1;
        const leftover = parseInt(l.slice(1)) % 100;
        position += mult * leftover;

        if (position < 0) {
            position = 100 + position;
        } else {
            position = position % 100;
        }

        if (position === 0) {
            zeros++;
        }
    }

    console.log(zeros);
}

function part2(lines: string[]): void {
    let position = 50;
    let zeros = 0;

    for (const l of lines) {
        let mult = (l[0] === 'L') ? -1 : 1;
        const prev = position;
        const clicks = parseInt(l.slice(1));
        zeros += Math.floor(clicks / 100);
        const leftover = clicks % 100;
        position += mult * leftover;

        if (prev !== 0 && (position <= 0 || position > 99)) {
            zeros++;
        }
        if (position < 0) {
            position = 100 + position;
        } else {
            position = position % 100;
        }
    }

    console.log(zeros);
}

part1(lines);
part2(lines);

part1(lines2);
part2(lines2);
